### PR TITLE
Add examples from Abdul-Razzak and Ghan 2000

### PR DESCRIPTION
**Pull request by claresinger**
https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/1999JD901161 

TODO:
- [x] Digitize and extra data from figures in paper, including the parameterization lines
- [x] Agree on which definition of activation to use. The paper uses the definition based on Smax > Scrit. --> Use both.
- [x] Use 50 SD per mode (50 bins in the paper)
- [x] Add figures 6-9? --> Not yet.
- [x] Fix nbviewer
- [x] Add nbviewer to README

Related: Move `kappa` calculation out of individual examples, https://github.com/atmos-cloud-sim-uj/PySDM/issues/789.

### CODE REVIEW

**Comment by slayoo** [comment]
thank you Clare!!!

**Comment by slayoo** [comment]
@claresinger @imdula @trontrytel this is amazing!

Just a few ideas before merging:
- [ ] let's separate data from code and move all the paper-retrieved curves data to separate files (.py or .csv) and let's name it intuitively like data_retrieved_from_paper.py
- [ ] the error bars are not visible in the `figs1-5.ipynb` notebook, if these are so small everywhere, let's remov the errorbar-plotting code... but they were not that small some time ago :)
- [ ] I'd strongly encourage to make run_parcel() work on SI units only, currently it expects rad2 in nanometres and N2 in cubic centimetres 
- [ ] let's convert all things like ``w = 0.5 # 0.5 m/s`` into ``w = 0.5 * si.m/si.s``
- [ ] let's perhaps use [namedtuples](https://docs.python.org/3.11/library/collections.html?highlight=namedtuple#collections.namedtuple) instead of tuples for returning multiple values (`output`, `output_attributes`, `aerosol`, `act_S`, `act_V`, `error`) from `run_parcel()` 
- [ ] ditto for aerosol specification - whenever the defined classes (in aerosol.py) only store data, let's use namedtuples (Pylint will anyhow ask us about it soon) 
- [ ] let's make the `aerosol_modes` dictionaries simpler with kappa as key and spectra as values? (currently nested, cumbersome to iterate through); if we need/want to handle multiple modes for same kappa, we could still simplify and use tuples of namedtuples instead of nested dictionaries, e.g. like
```python
from collections import namedtuple
ExternallyMixedMode = namedtuple("ExternallyMixedMode", ('kappa', 'spectrum'))
modes = (
    ExternallyMixedMode(kappa=.1, spectrum=Lognormal(...)),
    ExternallyMixedMode(kappa=.2, spectrum=Lognormal(...)),
)
```
will likely be more readable and the code that uses it will be shorter.

Thanks again!

**Comment by claresinger** [comment]
Thanks Sylwester! We definitely need to still have the option for two modes with same kappa, so the dictionary idea won't work, but I can do the namedtuples. I'll work on this and the other changes soon.

> * [ ]  let's make the `aerosol_modes` dictionaries simpler with kappa as key and spectra as values? (currently nested, cumbersome to iterate through); if we need/want to handle multiple modes for same kappa, we could still simplify and use tuples of namedtuples instead of nested dictionaries, e.g. like
> 
> ```python
> from collections import namedtuple
> ExternallyMixedMode = namedtuple("ExternallyMixedMode", ('kappa', 'spectrum'))
> modes = (
>     ExternallyMixedMode(kappa=.1, spectrum=Lognormal(...)),
>     ExternallyMixedMode(kappa=.2, spectrum=Lognormal(...)),
> )
> ```
> 
> will likely be more readable and the code that uses it will be shorter.

**Comment by claresinger** [comment]
Done
> * [x]  let's separate data from code and move all the paper-retrieved curves data to separate files (.py or .csv) and let's name it intuitively like data_retrieved_from_paper.py
> * [x]  I'd strongly encourage to make run_parcel() work on SI units only, currently it expects rad2 in nanometres and N2 in cubic centimetres
> * [x]  let's convert all things like `w = 0.5 # 0.5 m/s` into `w = 0.5 * si.m/si.s`

@trontrytel I *think* the error bars are just small when `n_SD=50`, but would be good to double check I haven't made a mistake in the calculation. If they are just small then either 1) we delete them or 2) we use fewer superdroplets so they are visible. I do think it's nice to show them.
> * [x]  the error bars are not visible in the `figs1-5.ipynb` notebook, if these are so small everywhere, let's remov the errorbar-plotting code... but they were not that small some time ago :)

@slayoo I might need some more help/guidance on how to implement these changes with namedtuples.
> * [ ]  let's perhaps use [namedtuples](https://docs.python.org/3.11/library/collections.html?highlight=namedtuple#collections.namedtuple) instead of tuples for returning multiple values (`output`, `output_attributes`, `aerosol`, `act_S`, `act_V`, `error`) from `run_parcel()`
> * [ ]  ditto for aerosol specification - whenever the defined classes (in aerosol.py) only store data, let's use namedtuples (Pylint will anyhow ask us about it soon)
> * [ ]  let's make the `aerosol_modes` dictionaries simpler with kappa as key and spectra as values? (currently nested, cumbersome to iterate through); if we need/want to handle multiple modes for same kappa, we could still simplify and use tuples of namedtuples instead of nested dictionaries, e.g. like

**Comment by slayoo** [comment]
@claresinger 

> @slayoo For the aerosols, maybe we can leave as is and then just clean this up altogether when we move all the aerosol.py functions into PySDM proper. What do you think?

Of course!

**Comment by trontrytel** [comment]
> @trontrytel I _think_ the error bars are just small when `n_SD=50`, but would be good to double check I haven't made a mistake in the calculation. If they are just small then either 1) we delete them or 2) we use fewer superdroplets so they are visible. I do think it's nice to show them.
> 
Not sure if I see the problem. I do see the error bars when I re-run the notebook locally. This is my Fig1 for example:

[fig_1.pdf](https://github.com/atmos-cloud-sim-uj/PySDM-examples/files/8209992/fig_1.pdf)

Assuming that this `Nj = np.asarray(output_attributes['n'])[j*n_sd_per_mode:(j+1)*n_sd_per_mode, -1]` correctly returns the multiplicities in each mode, everything looks great to me. 

I'm thinking that initially we were running those examples initialising with multiplicities that were not constant. So we would get a much bigger error bar if we were estimating it as a maximum multiplicity divided by the sum? Could that be the reason why now things look different than in our memories?

Anyhow, the PR looks excellent to me. Thank you @claresinger for getting it across the finish line with style and @imdula for figuring out the setup and starting the whole comparison. It is a great example of reproducible research.

**Review by trontrytel — APPROVED** [review]

File: PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    "    Fig2a_N2_obs, Fig2a_AF_obs, Fig2a_N2_param, Fig2a_AF_param,\n",
+    "    Fig2b_N2_obs, Fig2b_AF_obs, Fig2b_N2_param, Fig2b_AF_param,\n",
+    "    Fig3a_sol2_obs, Fig3a_AF_obs, Fig3a_sol2_param, Fig3a_AF_param,\n",
+    "    Fig3b_sol2_obs, Fig3b_AF_obs, Fig3b_sol2_param, Fig3b_AF_param,\n",
+    "    Fig4a_rad2_obs, Fig4a_AF_obs, Fig4a_rad2_param, Fig4a_AF_param,\n",
+    "    Fig4b_rad2_obs, Fig4b_AF_obs, Fig4b_rad2_param, Fig4b_AF_param,\n",
+    "    Fig5a_w_obs, Fig5a_AF_obs, Fig5a_w_param, Fig5a_AF_param,\n",
+    "    Fig5b_w_obs, Fig5b_AF_obs, Fig5b_w_param, Fig5b_AF_param\n",
+    ")\n",
+    "n_sd_per_mode = 10"
```
Comment (claresinger) [inline PySDM_examples/Abdul_Razzak_Ghan_2000/figs1-5.ipynb:46]: @trontrytel Sorry I should have clarified, I updated the branch to use much fewer SDs. So what you plotted is with `n_sd_per_mode = 10` and looks the same as it does for me :)

**Comment by slayoo** [comment]
Thank you @imdula, @claresinger & @trontrytel !
Merging. Release very soon.

**Pull request merged**